wireapp/wire-android-reloaded
# feat(conn-request): send connection request from other userprofile - pt.1 (AR-1486)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1486" title="AR-1486" target="_blank"><img alt="Subtask" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10816?size=medium" />AR-1486</a>  Modify the OtherUserProfileScreen to have a button to perform a connection request
  </td></table>
  <br />
 
















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?
https://wearezeta.atlassian.net/browse/AR-1486

### Issues

This PR allows sending a connection request, from the "other user profile" screen.

- Basic errors are handled, like: couldn't send request, couldn't load user profile data, etc.
- Call to Kalium UseCase to send a connection
- Navigation flow for success connection request
- Handles the case when the connection already was sent, and other statuses, so we can resend a connection to the target user depending if it was cancelled, or not.

What's not included and will be covered on next PR's
- Testing of the viewmodel
- Reloading/updating the search state so we can have a pending connection badge [see more](https://wearezeta.atlassian.net/browse/AR-1518)

### Testing

#### Test Coverage (Optional)


#### How to Test

You could search for a public user, and send a connection request to that contact. 
As a result, you should see the connection request on web, or the old client.

### Attachments (Optional)


https://user-images.githubusercontent.com/5806454/167852560-6b14e1cc-2b7c-4b1f-97f6-cc4924a3f305.mov


https://user-images.githubusercontent.com/5806454/167853132-ad86b86e-06ea-42e8-b291-d21574bb635b.mov


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
